### PR TITLE
Introduced two version support feature

### PR DESCRIPTION
**Pull request by hlts2**
Now that Vald can support multiple versions, clients should also be able to support multiple versions.
In this PR, I implemented a workflow to support multiple versions.
The main logic is implemented as a common workflow in [vald-client-ci](https://github.com/vdaas/vald-client-ci), so I implemented a call to it.

## Summary by CodeRabbit

- **New Features**
  - Updated release workflow to include licensing information, trigger configurations, and specific client settings.
  - Introduced a new workflow for syncing Vald using GitHub Actions with enhanced security measures and specific configurations.

- **Chores**
  - Updated `.gitignore` to exclude specific directories and files.
  - Enhanced the Makefile with new variables and targets to improve build and deployment processes.

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    tags:
+      - '*.*.*'
+      - 'v*.*.*'
+      - '*.*.*-*'
+      - 'v*.*.*-*'
+jobs:
+  release:
+    # TODO: change to the following before main merge.
+    # uses: vdaas/vald-client-ci/.github/workflows/_release.yaml@main
+    uses: vdaas/vald-client-ci/.github/workflows/_release.yaml@feat/add-common-workflows
```
Comment (hlts2) [inline .github/workflows/release.yaml:29]:
```suggestion
    # uses: vdaas/vald-client-ci/.github/workflows/release.yaml@main
    uses: vdaas/vald-client-ci/.github/workflows/release.yaml@feat/add-common-workflows
```

File: Makefile [hunk 2]
```diff
@@ -73,3 +89,73 @@ valdcli: $(TARGET_JAR)
...
+	curl -L https://raw.githubusercontent.com/rinx/word2vecjson/master/data/wordvecs1000.json -o $(TEST_DATASET_PATH)
+
+.PHONY: ci/deps/install
+## install deps for CI environment
+ci/deps/install: $(LEIN_PATH)
+
+.PHONY: ci/deps/update
+## update deps for CI environment
+ci/deps/update:
+	@echo "Nothing do be done"
```
Comment (hlts2) [inline Makefile:152]:
```suggestion
	@echo "Nothing to do"
```

**Comment by hlts2** [comment]
@coderabbitai review

File: .github/workflows/sync.yaml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# DO NOT EDIT
+name: "Sync Vald"
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: "*/10 * * * *"
```
Comment (hlts2) [inline .github/workflows/sync.yaml:21]:
```suggestion
    - cron: "0 */3 * * *"
```

**Review by kmrmt — APPROVED** [review]

**Pull request merged**